### PR TITLE
Issue 324. Updated map.js to handle errors when $.getJSON calls fail.…

### LINKED ISSUE #324: map section doesn't handle situation where api calls fail
if /unmatched-riders or /unmatched-drivers api calls fail (which it is doing now!) then the map section will be blank with no indication why

### PR DESCRIPTION
**Pull request by pagalvin**
… Updated UI to show a loading message and a "no data to map" type of message.

Hi - I see that someone already addressed this. please have a look at what I did anyway. Hopefully it's useful. 

In this case, "handle error" needs a bit more definition in terms of the actual handling. My code catches the error, logs details to the console and lets the user know that there's no data to show at this time (with an optimistic note that it's because no one has requested a ride at this time). 

This is my first attempt to contribute to this project.

Fixes #324

### CODE REVIEW

**Comment by richardwestenra** [comment]
@langdonms I'm gonna try to have a look over this in the next day or two, but I'd appreciate your feedback if you have time, as you built the map. It seems to be less urgent now that the server has been started back up and the map is working again, but it sounds like a good contribution to me.

**Comment by langdonms** [comment]
Hi Richard, at a conference next few days but will try to take a look
tonight, on mobile atm.
-langdon

File: pages/index.html [hunk 1]
```diff
@@ -335,7 +335,12 @@ <h2 class="bannerbox__title">Map of Participants</h2>
             <!-- Map CSS is inherited from Leaflet, but need to set a minimum height -->
             <div class="bannerbox__content">
                 <p>This map shows voters who still need a ride and available volunteer drivers.</p>
-                <div id="map" style="height: 400px; width: 100%;"></div>
+                <p id="mapLoadingMessage">Loading requested rides and volunteers...</p>
+                <p id="noMapDataAvailable" style="display:none">
+                    We don't have any participants to map at the moment. As soon as someone (maybe you!) reqeuests a ride 
+                    or offers a ride, they will show up here.
+                </div>
```
Comment (richardwestenra) [inline pages/index.html]: This `</div>` should be a `</p>`, otherwise it breaks the page markup! :(

**Comment by richardwestenra** [comment]
Hi @pagalvin, it turns out there was already a preexisting active PR for this issue, and it solved the problem too so I am closing this PR. Many thanks for your contribution, we really appreciate it.]

If you don't want this to happen again, I recommend checking in the comment sections under reviews to check whether an issue has been referenced from another PR, which you can do in a PR description or commit message, using a # symbol plus the PR ID number (as I did in the preceding comment).

Thanks again!

**Pull request closed without merging**